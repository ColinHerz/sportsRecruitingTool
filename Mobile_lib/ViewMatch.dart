import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'Session.dart';
import 'dart:async';
import 'dart:convert';

class ViewMatch extends StatefulWidget{
  final String matchID;

  ViewMatch(this.matchID);

  _ViewMatch createState() => _ViewMatch(matchID);
}
enum choose{yes, no}
class _ViewMatch extends State<ViewMatch>{
  final String matchID;

  List<dynamic> subTotalScores = new List<String>();
  //List<String> golf = new List<String>();
  String bagID;
  String datePlayed;
  String matchName;
  String bagName;
  String course;
  int totalScore;
  List<Widget> displayMatch = new List<Widget>();
  final _scoreController = new TextEditingController();
  final _puttController = new TextEditingController();
  choose currVal1;
  choose currVal2;

  _ViewMatch(this.matchID);

  Future<void> getMatch() async {
    final http.Response _response = await http.get(
      'http://localhost3000.us-east-2.elasticbeanstalk.com/api/golf/getGolfMatch/$matchID',
      headers: headers,
    );

    if (_response.statusCode == 200) {
      var parsedJson = json.decode(_response.body);
      print(parsedJson);
      setState(() {
        matchName = parsedJson['nameOfRound'];
        datePlayed = parsedJson['datePlayed'];
        subTotalScores = parsedJson['subTotalScores'];
        totalScore = parsedJson['totalScore'];
        bagID = parsedJson['GolfBagUsed'];
        course = parsedJson['coursePlayed'];
      });
      //golf = parsedJson['golfMatch'];
      getBag(jsonEncode(<String, String>{
        'golfBag': bagID,
      }));
    }
    else {
      print(_response.statusCode);
      print(_response.body);
    }
  }
  Future<void> getBag(dynamic info) async {
    final http.Response _response = await http.post(
        'http://localhost3000.us-east-2.elasticbeanstalk.com/api/golf/getGolfBag',
        headers: headers,
        body: info
    );

    if (_response.statusCode == 200) {
      var parsedJson = json.decode(_response.body);
      setState(() {
        bagName = parsedJson['bagName'];
      });
    }
    else {
      print(_response.statusCode);
      print(_response.body);
    }
  }

  List<Widget> display(){
    displayMatch = new List<Widget>();

    displayMatch.add(Text("$matchName", style: largeStyle));
    displayMatch.add(Text("$course", style: TextStyle(fontSize: 22, fontWeight: FontWeight.bold)));
    displayMatch.add(Padding(padding: EdgeInsets.fromLTRB(0, 0, 0, 10)));
    displayMatch.add(Text("Date Played: $datePlayed", style: TextStyle(fontSize: 20)));
    displayMatch.add(Padding(padding: EdgeInsets.fromLTRB(0, 0, 0, 10)));
    displayMatch.add(Text("Bag Used: $bagName", style: TextStyle(fontSize: 20)));
    displayMatch.add(Padding(padding: EdgeInsets.fromLTRB(0, 10, 0, 10)));
    if (subTotalScores.length == 0)
      displayMatch.add(Text("Total Score: 0", style: TextStyle(fontSize: 25, fontWeight: FontWeight.bold)));
    else
      displayMatch.add(Text("Total Score: ", style: TextStyle(fontSize: 25, fontWeight: FontWeight.bold)));

    displayMatch.add(Padding(padding: EdgeInsets.fromLTRB(0, 0, 0, 20)));
    displayMatch.add(Text("New Score", style: TextStyle(fontSize: 20)));

    displayMatch.add(TextField(
        controller: _scoreController,
        decoration: InputDecoration(
          border: OutlineInputBorder(),
          labelText: 'Score',
        )
    ));

    displayMatch.add(TextField(
        controller: _puttController,
        decoration: InputDecoration(
          border: OutlineInputBorder(),
          labelText: 'Number of Putts',
        )
    ));

    displayMatch.add(Text("Fairway Hit:", style: TextStyle(fontSize: 20)));

    displayMatch.add(
      ListTile(
        title: const Text('Yes'),
        leading: Radio(
          value: choose.yes,
          groupValue: currVal1,
          onChanged: (choose value) {
            setState(() { currVal1 = value; });
          },
        ),
      ),
    );
    displayMatch.add(
      ListTile(
        title: const Text('No'),
        leading: Radio(
          value: choose.no,
          groupValue: currVal1,
          onChanged: (choose value) {
            setState(() { currVal1 = value; });
          },
        ),
      ),
    );

    displayMatch.add(Text("Green in Regulation:", style: TextStyle(fontSize: 20)));

    displayMatch.add(
      ListTile(
        title: const Text('Yes'),
        leading: Radio(
          value: choose.yes,
          groupValue: currVal2,
          onChanged: (choose value) {
            setState(() { currVal2 = value; });
          },
        ),
      ),
    );
    displayMatch.add(
      ListTile(
        title: const Text('No'),
        leading: Radio(
          value: choose.no,
          groupValue: currVal2,
          onChanged: (choose value) {
            setState(() { currVal2 = value; });
          },
        ),
      ),
    );
    displayMatch.add(Row(
      mainAxisAlignment: MainAxisAlignment.center,
      children: <Widget>[
        Center(child: ButtonTheme(
            child: RaisedButton(
              onPressed: (){
                Navigator.pop(context);
              },
              child: Text("Add score"),
            )
        )
        ),
        Center(child: RaisedButton(
            onPressed: (){
              Navigator.pop(context);
            },
            child: Text('Back')
        )
        )
      ],
    ));


    return displayMatch;
  }

  void initState(){
    super.initState();
    getMatch();
  }

  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('View Match'),
      ),
      body: ListView(
          children: display()
      ),
    );
  }
}
