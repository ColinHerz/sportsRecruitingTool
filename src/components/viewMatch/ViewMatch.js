import apiCall from "../../api/api.js";
import { Link } from "react-router-dom";
import React, { useEffect, useState } from "react";
import { useForm } from "react-hook-form";
import { useParams } from "react-router-dom";

import "./viewMatch.scss";

const clubs = [
	``,
	`Wood`,
	`Iron`,
	`Wedge`,
	`Putter`
];

const ViewMatch = props => {
	const { mid } = useParams();
	const { register, handleSubmit, errors } = useForm();

	const [updateMatch, setUpdateMatch] = useState(true);
	const [updateEvents, setUpdateEvents] = useState(true);
	const [showError, setShowError] = useState(false);

	const [matchInfo, setMatchInfo] = useState({});
	const [golfBag, setGolfBag] = useState({});

	const [showAddScore, setShowAddScore] = useState(false);

	const [events, setEvents] = useState([]);
	const [showEventPicker, setShowEventPicker] = useState(false);
	const [matchSendSuccess, setMatchSendSuccess] = useState(false);

	useEffect(() => {
		const getBag = id => {
			apiCall(
				{
					endpoint: `/golf/getGolfBag`,
					type: `POST`,
					body: {
						golfBag: id
					}
				},
				data => {
					if (data.status !==  200) {
						setShowError(true);
						return;
					}

					setGolfBag(JSON.parse(data.response));
				},
				setShowErrorMessage
			);
		};

		if (updateMatch) {
			apiCall(
				{
					endpoint: `/golf/getGolfMatch/${mid}`,
					type: `GET`
				},
				data => {
					if (data.status !== 200) {
						setShowError(true);
						return;
					}

					const response = JSON.parse(data.response);

					setMatchInfo(response);
					getBag(response.GolfBagUsed);
					setUpdateMatch(false);
				},
				setShowErrorMessage
			);
		}
	}, [updateMatch, mid]);

	useEffect(() => {
		if (updateEvents) {
			apiCall(
				{
					endpoint: `/golf/getMyEvents`,
					type: `GET`
				},
				data => {
					if (data.status !== 200) {
						setShowError(true);
						return;
					}

					setEvents(JSON.parse(data.response));
					setUpdateEvents(false);
				},
				setShowErrorMessage
			);
		}
	}, [updateEvents]);

	const toggleAddScore = event => {
		event.preventDefault();
		setShowAddScore(!showAddScore);
	};

	const onSubmit = info => {
		setShowAddScore(false);
		const body = Object.assign({}, info);
		body.golfMatch = mid;

		apiCall(
			{
				endpoint: `/golf/createHoleScore`,
				type: `POST`,
				body: body
			},
			data => {
				if (data.status !== 200) {
					setShowError(true);
					return;
				}

				setUpdateMatch(true);
			},
			setShowErrorMessage
		);
	};

	const toggleEventPicker = event => {
		event.preventDefault();
		setShowEventPicker(!showEventPicker);
	};

	const sendMatch = info => {
		apiCall(
			{
				endpoint: `/golf/postEventScore`,
				type: `POST`,
				body: {
					event: info.event,
					golfMatch: mid
				}
			},
			data => {
				if (data.status !== 200) {
					setShowError(true);
					return;
				}

				setMatchSendSuccess(true);
				setShowEventPicker(false);
			},
			setShowErrorMessage
		);
	};

	const cancelSendMatch = event => {
		event.preventDefault();

		setShowEventPicker(false);
	};

	const setShowErrorMessage = () => {
		setShowError(true);
	};

	return (
		<main id="view-match">
			{
				showError ?
					<p className="error-text">Something went wrong. Please try again later.</p>:
					null
			}

			{
				updateMatch ?
					<p>Loading...</p>:
					<React.Fragment>
						<h2>{matchInfo.nameOfRound}</h2>

						<h3>{matchInfo.coursePlayed}</h3>

						<button id="send-match-btn" onClick={toggleEventPicker}>Send Match to an Event</button>

						{
							showEventPicker ?
								<form onSubmit={handleSubmit(sendMatch)}>
									<label>
										Select an Event:
										<select
											name="event"
											ref={
												register({
													required: true
												})
											}
										>
											<option disabled>Please select an Event</option>

											{
												events.map(event => {
													return (
														<option
															key={event._id}
															value={event._id}
														>
															{event.eventName}
														</option>
													);
												})
											}
										</select>
									</label>

									<input type="submit" value="Send" />
									<button onClick={cancelSendMatch}>Cancel</button>
								</form>:
								null
						}

						{
							matchSendSuccess ?
								<p>Match sent successfully.</p>:
								null
						}

						<p><span className="bolder">Date Played:</span> {(new Date(matchInfo.datePlayed)).toDateString()}</p>

						<p><span className="bolder">Bag Used:</span> <Link to={`/profile/bag/${matchInfo.GolfBagUsed}`}>{golfBag.bagName}</Link></p>

						<p><span className="bolder">Total Score:</span> {matchInfo.totalScore}</p>

						<h4>Scores</h4>

						<p>Click on a score for more info.</p>

						<button id="add-score-btn" onClick={toggleAddScore}>Add Score</button>

						{
							showAddScore ?
								<form onSubmit={handleSubmit(onSubmit)}>
									<label className="wide-label">
										Score
										<input
											type="number"
											name="score"
											placeholder="3"
											ref={
												register({
													required: true
												})
											}
										/>
									</label>
									{errors.score && <p className="error-text">A score is required.</p>}

									<label className="wide-label">
										Number of Putts
										<input
											type="number"
											name="numberOfPutts"
											placeholder="2"
											ref={
												register({
													required: true
												})
											}
										/>
									</label>
									{errors.numberOfPutts && <p className="error-text">A putt count is required.</p>}

									<h5>Fairway Hit</h5>
									<div>
										<input
											type="radio"
											name="fairwayHit"
											id="yes-fairway"
											value={true}
											ref={
												register({
													required: true
												})
											}
										/>
										<label htmlFor="yes-fairway">Yes</label>
									</div>

									<div>
										<input
											type="radio"
											name="fairwayHit"
											id="no-fairway"
											value={false}
											ref={
												register({
													required: true
												})
											}
										/>
										<label htmlFor="no-fairway">No</label>
									</div>
									{errors.fairwayHit && <p className="error-text">Fairway hit is required.</p>}

									<h5>Green in Regulation</h5>
									<div>
										<input
											type="radio"
											name="greenInRegulation"
											id="yes-green"
											value={true}
											ref={
												register({
													required: true
												})
											}
										/>
										<label htmlFor="yes-green">Yes</label>
									</div>

									<div>
										<input
											type="radio"
											name="greenInRegulation"
											id="no-green"
											value={false}
											ref={
												register({
													required: true
												})
											}
										/>
										<label htmlFor="no-green">No</label>
									</div>
									{errors.greenInRegulation && <p className="error-text">Green in regulation is required.</p>}

									<h5>Clubs Used</h5>
									{
										golfBag.golfClub.map(club => (
											<div key={club._id}>
												<input
													type="checkbox"
													name="clubsUsed"
													id={club._id}
													value={club._id}
													ref={
														register({
															required: true
														})
													}
												/>
												<label htmlFor={club._id}>{club.clubName} ({clubs[club.clubType]})</label>
											</div>
										))
									}
									{errors.clubsUsed && <p className="error-text">Clubs used is required.</p>}

									<div id="submission-btns">
										<input type="submit" value="Done" />
										<button onClick={toggleAddScore}>Cancel</button>
									</div>
								</form>:
								null
						}

						<ol id="scores-grid">
							{
								matchInfo.golfMatch.map(hole => {
									return(
										<li key={hole._id}>
											<Link to={`/match/${mid}/${hole._id}/`}>{hole.score}</Link>
										</li>
									);
								})
							}
						</ol>
					</React.Fragment>
			}
		</main>
	);
};

export default ViewMatch;
