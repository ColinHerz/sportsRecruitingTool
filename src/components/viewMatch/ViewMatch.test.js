/* eslint-env jest */
import { MemoryRouter } from "react-router-dom";
import React from "react";
import renderer from "react-test-renderer";

import ViewMatch from "./ViewMatch.js";

const genProps = () => {
	return {
		user: {}
	};
};

describe(`Rendering`, () => {
	it(`Renders beginning page correctly`, () => {
		const props = genProps();

		const component = renderer.create(
			<MemoryRouter>
				<ViewMatch {...props} />
			</MemoryRouter>
		);
		const tree = component.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
