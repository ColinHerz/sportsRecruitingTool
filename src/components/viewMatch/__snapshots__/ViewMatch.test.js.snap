// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Renders beginning page correctly 1`] = `
<main
  id="view-match"
>
  <p>
    Loading...
  </p>
</main>
`;
